prefill_haalcentraalhr/models.py

```diff
@@ -32,9 +33,13 @@ class HaalCentraalHRConfig(SingletonModel):
     class Meta:
         verbose_name = _("Haal Centraal HR configuration")
 
-    def build_client(self) -> PreRequestZGWClient | None:
+    def build_client(self, submission: Submission = None) -> PreRequestZGWClient | None:
         if not self.service:
             logger.info("No service configured for Haal Centraal HR.")
             return
 
-        return self.service.build_client()
+        client = self.service.build_client()
+        if submission:
+            client.context = PreRequestClientContext(submission=submission)
+
+        return client
```

Comment:
I think the type hints state that `submission` may be `None`, so perhaps you can/should always set the context and can remove the `if submission` check?